nixos/modules/services/web-servers/nginx/default.nix

```diff
@@ -638,9 +637,15 @@ in
       preStart =
         ''
         ${cfg.preStart}
-        ${cfg.package}/bin/nginx -c ${configFile} -p ${cfg.stateDir} -t
+        test -d /tmp/.spool/nginx/logs || mkdir -p -m 750 /tmp/.spool/nginx/logs
+        ${cfg.package}/bin/nginx -c ${configFile} -p /tmp/.spool/nginx -t
```

Comment:
Instead of fixing permissions of files created by "nginx -t", we can run "nginx -t" with temp spool dir.